DEVELOPING.rst

```diff
@@ -8,461 +8,91 @@
-
-   $ docker exec -i -t opendatacernch-web-1 cernopendata fixtures records -f /code/cernopendata/modules/fixtures/data/records/opera-ed-datasets.json --mode replace
-
-Working with files
-------------------
-
-If you are working with serving file assets, please note that ``web-files``
-container may loose XRootD connection to EOS if you change networks or resume
-your laptop from the deep sleep. In this case it may be necessary to restart the
-``web`` and ``web-files`` containers:
-
-.. code-block:: console
-
-   $ docker compose -f docker-compose-dev.yml restart web web-files
-
-Working with proxy
-------------------
-
-If you are working in a production environment and you need to delete the proxy
-cache content, you can run:
-
-.. code-block:: console
-
-   $ docker exec opendatacernch-nginx-1 find /var/cache/nginx -type f -delete
-
-Working with UI packages
-------------------------
-
-When working on UI packages that have JavaScript and CSS files, you can have
-"live editing" by running the following command on a new terminal:
-
-.. code-block:: console
-
-   $ docker exec -i -t opendatacernch_web_1 cernopendata webpack run start
-
-Keep in mind that you need to recreate the ``package.json`` when adding or
-removing dependencies:
-
-.. code-block:: console
-
-   $ docker exec -i -t opendatacernch_web_1 cernopendata webpack clean create
-
-Working with iSpy visualizer
-----------------------------
-
-CSS dependencies which are needed for iSpy CMS visualizer are sandboxed in order
-to make it compatible with ``Semantic UI``. This was achieved by:
-
-* Wrapping all the ``Bootstrap`` html with a ``<div class="bootstrap-ispy">``
-* Prefixing all the css classes of ``Bootstrap`` framework and custom ispy css file with ``bootstrap-ispy`` class.
-* As a result ``Bootstrap`` css can be used inside a div with ``bootstrap-ispy`` class without any conflicts with ``Semantic UI``.
-
-Procedure to prefix css files with ``bootstrap-ispy`` class:
-
-* Download unminified version (CMS visualizer currently uses Bootstrap v3.3.1) of the ``Bootstrap`` framework from the official website (usually it's bootstrap.css file)
-* Install LESS preprocessor locally: ``npm install -g less``
-* Create a file ``prefix-bootstrap.less`` which contains the following:
-
-.. code-block:: console
-
-   .bootstrap-ispy {
-      @import (less) 'bootstrap.css';
-   }
-
-* Preprocess css file with LESS to generate a new prefixed file (it will create ``bootstrap-prefixed.css`` file):
+For a quickstart guide, do the following:
 
 .. code-block:: console
 
-   lessc prefix-bootstrap.less bootstrap-prefixed.css
+  $ # Checkout this repository
+  $ git clone https://github.com/cernopendata/opendata.cern.ch.git
+  $ # Checkout the module with the portal
+  $ git clone https://github.com/cernopendata/cernopendata-portal.git
+  $ # Move to the directory of the content
+  $ cd opendata.cern.ch
+  $ # The command line is a bit long: let's create an alias to avoid typing all of that
```

Comment:
Coming back to this resolved comment.  (BTW it may be good to keep them opened for easy search/find.) 

We could take advantage of [includes and overrides](https://docs.docker.com/compose/multiple-compose-files/include/#include-and-overrides) to create the following file `docker-compose.yml` file in this repository:

```yaml
include:
  - path:
      - ../cernopendata-portal/docker-compose.yml
      - docker-compose-content.yml
```

This should allow people to type directly the usual `docker compose` commands they are already used to, such as:

```console
$ docker compose pull
$ docker compose up
$ docker exec -i -t opendatacernch-web-1 cernopendata fixtures records \
              --mode insert-or-replace \
              -f /content/data/records/cms-primary-datasets.json
```

This should be more comfortable user-experience-wise when compared to introducing an unusual `my_docker` alias that may look a bit strange.
 
(I would also rename `docker-compose-content.yml` to something like `docker-compose-override.yml` to make its purpose very clear.)

Reply:
@psaiz Unresolving this conversation because `my_docker` still seems to be there?

Reply:
Please, submit a suggestion with the exact changes that you want. That will speed things up.

Reply:
It is not possible to send suggestion in the GitHub style, because I'm proposing to alter files. If my above comment was not clear, then this is what I was proposing:

```bash
echo "include:" > docker-compose.yml
echo " - path:" >> docker-compose.yml
echo "   - ../cernopendata-portal/docker-compose.yml" >> docker-compose.yml
echo "   - docker-compose-override.yml" >> docker-compose.yml
git add docker-compose.yml
git mv docker-compose-content.yml docker-compose-override.yml
git commit --amend --no-edit
```

The help instructions can simply say:

```console
$ docker compose pull
$ docker compose up
$  docker exec -i -t opendatacernch-web-1 /code/scripts/populate-instance.sh --skip-records --skip-glossary --skip-docs
$ docker exec -i -t opendatacernch-web-1 cernopendata fixtures records \
              --mode insert-or-replace \
              -f /content/data/records/cms-primary-datasets.json
```

This will do for now for the merge to happen. 

I'll be amending the documenation page considerably after merge, as I mentioned elsewhere (e.g. [here](https://github.com/cernopendata/opendata.cern.ch/pull/3590#discussion_r1714983018) and [here](https://github.com/cernopendata/opendata.cern.ch/pull/3590#discussion_r1746732736)) which were closed by not addressed. It will be simpler if I take care of that.


P.S.  Note that you'll need to also rebase against latest `master`.

Reply:
Done, and rebased